### PR TITLE
refactor: add distributed tracing context activator helper in trace_utils

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
As per [APMPY-455](https://datadoghq.atlassian.net/jira/software/projects/APMPY/boards/136?selectedIssue=APMPY-455), a helper function was added in trace_utils.py to extract a context from request headers and activate (if enabled) that distributed tracing context.

## Checklist
- [ ] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: tests/tracer/test_trace_utils.py [hunk 1]
```diff
@@ -160,3 +161,17 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
...
+    pin = Pin()
+    int_config.myint["distributed_tracing_enabled"] = True
+    headers = {
+        HTTP_HEADER_PARENT_ID: "12345",
+        HTTP_HEADER_TRACE_ID: "678910",
+    }
+    trace_utils.activate_distributed_tracing_context(pin, int_config.myint, headers)
+    context = pin.tracer.context_provider.active()
+
+    assert context._parent_trace_id == 678910
```
Comment (Kyle-Verhoog) [inline tests/tracer/test_trace_utils.py:176]:
```suggestion
    assert context.trace_id == 678910
```

File: tests/tracer/test_trace_utils.py [hunk 2]
```diff
@@ -160,3 +161,17 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
...
+    assert context._parent_span_id == 12345
```
Comment (Kyle-Verhoog) [inline tests/tracer/test_trace_utils.py:177]:
```suggestion
    assert context.span_id == 12345
```

File: ddtrace/contrib/trace_utils.py [hunk 3]
```diff
@@ -175,3 +176,17 @@ def set_http_meta(
 
     if response_headers is not None:
         store_response_headers(dict(response_headers), span, integration_config)
+
+
+def activate_distributed_tracing_context(pin, int_config, request_headers):
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/trace_utils.py:181]:
```suggestion
def activate_distributed_tracing_context(tracer, int_config, request_headers):
```

I don't think we use `Pin`s consistently enough throughout the library for the tracer instance (see Pyramid for example). So I think we can just pass a tracer reference directly.

Comment (Kyle-Verhoog) [inline ddtrace/contrib/trace_utils.py:181]:
```suggestion
def activate_distributed_headers(int_config, request_headers):
```

File: tests/tracer/test_trace_utils.py [hunk 4]
```diff
@@ -160,3 +161,17 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
         mock_log.exception.assert_called_once_with(*log_call)
     else:
         mock_log.exception.assert_not_called()
+
+
+def test_activate_distributed_tracing_context(int_config):
+    pin = Pin()
+    int_config.myint["distributed_tracing_enabled"] = True
```
Comment (Kyle-Verhoog) [inline tests/tracer/test_trace_utils.py:168]: Would be good to have test cases for when distributed tracing is disabled

File: tests/tracer/test_trace_utils.py [hunk 5]
```diff
@@ -160,3 +161,17 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
...
+    headers = {
```
Comment (Kyle-Verhoog) [inline tests/tracer/test_trace_utils.py:169]: Also another test case for when headers aren't present

File: tests/tracer/test_trace_utils.py (hunk 1 above)
Comment (Yun-Kim) [reply tests/tracer/test_trace_utils.py:177]: Ahh good catch, I only saw the private attributes 😨

File: tests/tracer/test_trace_utils.py [hunk 6]
```diff
@@ -160,3 +162,42 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
         mock_log.exception.assert_called_once_with(*log_call)
     else:
         mock_log.exception.assert_not_called()
+
+
+def test_activate_distributed_headers_enabled(int_config):
+    pin = Pin(tracer=Tracer())
```
Comment (Kyle-Verhoog) [inline tests/tracer/test_trace_utils.py:168]:
```suggestion
```

don't actually need a pin in these tests anymore 🙂

Comment (Yun-Kim) [reply tests/tracer/test_trace_utils.py]: Thanks for the catch 🤒

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**